queries.sql

```diff
@@ -7,4 +7,56 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
+
+/* update the animals table by setting the species column to unspecified */
+-- Transaction 2
+BEGIN;
+update animals set species = 'unspecified';
+SELECT * from animals;
+ROLLBACK;
+SELECT * from animals;
+
+-- Transaction 2 
+BEGIN;
+update animals set species = 'digimon' where name like '%mon';
+update animals set species = 'pokemon' where species IS null;
+SELECT * from animals;
+COMMIT;
+SELECT * from animals;
+
+/* Delete all records in the animals table transaction 5 */
+-- Transaction 3
+BEGIN;
+DELETE from animals;
+SELECT * from animals;
+ROLLBACK;
+SELECT * from animals;
+
+-- Transaction 4
+BEGIN;
+DELETE from animals WHERE date_of_birth > '01-01-2022';
+SAVEPOINT dob;
+UPDATE animals SET weight_kg =weight_kg * -1;
+ROLLBACK TO dob;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * from animals;
+
+/* How many animals are there? */
+SELECT COUNT(*) from animals;
+
+/* How many animals have never tried to escape? */
+SELECT COUNT(*) from animals WHERE escape_attempts = 0;
+
+/* What is the average weight of animals? */
+SELECT AVG(weight_kg) from animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT neutered, count(neutered) as result FROM animals GROUP BY neutered ORDER BY result DESC LIMIT 1;
```

Comment:
- [ ] Good job but here you are counting the `neutered` animals but the question is about `escape_attempts` like who escapes the most, neutered or not neutered animals. kindly consider writing the right query for that.
hint: `AVG(escape_attempts)` GROUP BY neutered;